app/services/service_provider_seeder.rb

```diff
@@ -55,10 +56,16 @@ def write_service_provider?(config)
     return true if rails_env != 'production'
 
     restrict_env = config['restrict_to_deploy_env']
+    in_prod = deploy_env == 'prod'
+    in_sandbox = !%w[prod staging].include?(deploy_env)
+    in_staging = deploy_env == 'staging'
 
-    is_production_or_has_a_restriction = (deploy_env == 'prod' || restrict_env.present?)
+    return true if restrict_env == 'prod' && in_prod
```

Comment:
Please double-check me here but I'm 99% sure this is the same as the previous code with the exception of line 65 - structuring it this way felt a bit more readable to me.